libs/datamodel/core/tests/attributes/default_negative.rs

```diff
@@ -669,18 +669,296 @@ fn must_error_on_dbgenerated_default_on_non_native_type_on_mongodb() {
 
         model User {
             id Int @id @map("_id")
-            nickname String @default(dbgenerated())
+            nickname String @default(auto())
         }
     "#;
 
     let error = datamodel::parse_schema(schema).map(drop).unwrap_err();
 
     let expected = expect![[r#"
-        [1;91merror[0m: [1mError validating field 'nickname': MongoDB `@default(dbgenerated())` fields must have a native type annotation.[0m
+        [1;91merror[0m: [1mError validating field 'nickname': MongoDB `@default(auto())` fields must have `ObjectId` native type and use the `@id` attribute.[0m
           [1;94m-->[0m  [4mschema.prisma:14[0m
         [1;94m   | [0m
         [1;94m13 | [0m            id Int @id @map("_id")
-        [1;94m14 | [0m            [1;91mnickname String @default(dbgenerated())[0m
+        [1;94m14 | [0m            [1;91mnickname String @default(auto())[0m
+        [1;94m15 | [0m        }
+        [1;94m   | [0m
+    "#]];
+
+    expected.assert_eq(&error)
+}
+
+#[test]
+fn must_error_on_auto_default_on_non_object_id_on_mongodb() {
+    let schema = r#"
+        datasource db {
+            provider = "mongodb"
+            url = "mongodb://"
+        }
+
+        generator client {
+            provider = "prisma-client-js"
+            previewFeatures = ["mongoDb"]
+        }
+
+        model User {
+            id Int @id @map("_id") @default(auto()) @db.Int
+            nickname String
+        }
+    "#;
+
+    let error = datamodel::parse_schema(schema).map(drop).unwrap_err();
+
+    let expected = expect![[r#"
+        [1;91merror[0m: [1mError validating field 'id': MongoDB `@default(auto())` fields must have `ObjectId` native type and use the `@id` attribute.[0m
+          [1;94m-->[0m  [4mschema.prisma:13[0m
+        [1;94m   | [0m
+        [1;94m12 | [0m        model User {
+        [1;94m13 | [0m            [1;91mid Int @id @map("_id") @default(auto()) @db.Int[0m
+        [1;94m14 | [0m            nickname String
+        [1;94m   | [0m
+    "#]];
+
+    expected.assert_eq(&error)
+}
+
+#[test]
+fn must_error_on_auto_default_on_mongodb_composite() {
+    let schema = r#"
+        datasource db {
+            provider = "mongodb"
+            url = "mongodb://"
+        }
+
+        generator client {
+            provider = "prisma-client-js"
+            previewFeatures = ["mongoDb"]
+        }
+
+        model User {
+            id   String @id @map("_id") @default(auto()) @db.ObjectId
+            meow Meow
+        }
+
+        type Meow {
+            id String @default(auto())
+        }
+    "#;
+
+    let error = datamodel::parse_schema(schema).map(drop).unwrap_err();
+
+    let expected = expect![[r#"
+        [1;91merror[0m: [1mError parsing attribute "@default": The function `auto()` cannot be used on fields of type `String`.[0m
```

Comment:
I think this one is not the error we would expect. I would have thought it triggers the error about auto in composite types. The message is also misleading, because we do support auto on string fields, just not in composite types.

Reply:
Fixed in [`685b087` (#2648)](https://github.com/prisma/prisma-engines/pull/2648/commits/685b087aac0c07a9762022a011f161d501bb8292)